pkg/controller/wip/controller_test.go

```diff
@@ -185,6 +260,248 @@ func TestReconcileInstanceHappyLand(t *testing.T) {
+	}
+	if e, a := "ReferencesNonexistentServiceClass", updateActionObject.Status.Conditions[0].Reason; e != a {
+		t.Fatalf("Unexpected condition reason: expected %v, got %v", e, a)
+	}
+	close(stopCh)
+}
+
+func TestReconcileBindingWithParameters(t *testing.T) {
+	_, fakeCatalogClient, fakeBrokerCatalog, _, fakeBindingClient, testController, sharedInformers, stopCh := newTestController(t)
+
+	fakeBrokerCatalog.RetCatalog = testCatalog
+
+	sharedInformers.Brokers().Informer().GetStore().Add(broker)
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(serviceClass)
+	instance := &v1alpha1.Instance{
+		ObjectMeta: v1.ObjectMeta{Name: "test-instance", Namespace: "test-ns"},
+		Spec: v1alpha1.InstanceSpec{
+			ServiceClassName: "test-serviceclass",
+			PlanName:         "default",
+		},
+	}
+	sharedInformers.Instances().Informer().GetStore().Add(instance)
+
+	binding := &v1alpha1.Binding{
+		ObjectMeta: v1.ObjectMeta{Name: "test-binding", Namespace: "test-ns"},
+		Spec: v1alpha1.BindingSpec{
+			InstanceRef: v1.ObjectReference{Name: "test-instance", Namespace: "test-ns"},
+		},
+	}
+
+	parameters := bindingParameters{Name: "test-param"}
+	parameters.Args = append(parameters.Args, "first-arg")
+	parameters.Args = append(parameters.Args, "second-arg")
+	b, err := json.Marshal(parameters)
+	if err != nil {
+		t.Fatalf("Failed to marshal parameters %v : %v", parameters, err)
+	}
+	binding.Spec.Parameters.Raw = b
+
+	testController.reconcileBinding(binding)
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 1, len(actions); e != a {
+		t.Logf("%+v\n", actions)
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+
+	// There should only be one action that says binding was created
+	updateAction := actions[0].(core.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[0]; expected %v, got %v", e, a)
+	}
+	updateObject := updateAction.GetObject().(*v1alpha1.Binding)
+	if e, a := "test-binding", updateObject.Name; e != a {
+		t.Fatalf("Unexpected name of binding created: expected %v, got %v", e, a)
+	}
+	if e, a := 1, len(updateObject.Status.Conditions); e != a {
+		t.Fatalf("Unexpected number of conditions: expected %v, got %v", e, a)
+	}
+	if e, a := "InjectedBindResult", updateObject.Status.Conditions[0].Reason; e != a {
+		t.Fatalf("Unexpected condition reason: expected %v, got %v", e, a)
+	}
+
+	// Verify parameters are what we'd expect them to be, basically name, array with two values in it.
+	if len(updateObject.Spec.Parameters.Raw) == 0 {
+		t.Fatalf("Parameters was unexpectedly empty")
+	}
+	// TODO(vaikas): Implement the storing logic in the fake Binding Client so that it stores
+	// something meaningful there. For now, it just stores a struct making the validation a bit
+	// wonky.
+	if _, ok := fakeBindingClient.Bindings[":"]; !ok {
+		t.Fatalf("Did not find the created Binding in fakeInstanceBinding after creation")
```

Comment:
Hm, perhaps I misunderstood what this does then:
https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/brokerapi/fake/fake.go#L182
Seems like it's not storing anything useful there, but ??